Source/WebCore/rendering/RenderObject.cpp

```diff
@@ -2349,6 +2350,22 @@ Vector<FloatRect> RenderObject::clientBorderAndTextRects(const SimpleRange& rang
     return borderAndTextRects(range, CoordinateSpace::Client, { });
 }
 
+ScrollAnchoringController* RenderObject::scrollAnchoringController() const
+{
+    if (hasLayer()) {
+        if (auto* scrollableArea = downcast<RenderLayerModelObject>(*this).layer()->scrollableArea())
+            return scrollableArea->scrollAnchoringController();
+    }
+    for (auto* renderer = this; renderer; renderer = renderer->parent()) {
```

Comment:
This loop would be faster if written in terms of `RenderLayer`. So call `enclosingLayer()` and then walk up the layer parent chain.

Reply:
Fixed.